designs/2019-config-simplification/README.md

```diff
@@ -0,0 +1,544 @@
+- Start Date: 2019-01-20
+- RFC PR: TBD
+- Authors: Nicholas C. Zakas (@nzakas)
+
+# Config File Simplification
+
+## Summary
+
+This proposal provides a way to simplify configuration of ESLint through a new configuration file format. The configuration file format is written in JavaScript and removes several existing configuration keys in favor of allowing the user to manually create them.
+
+## Motivation
+
+The ESLint configuration file format (`.eslintrc`) hasn't changed much since ESLint was first released. As we have continued to add new features, the configuration file format has continued to be extended and evolve into a state where it is the most complicated part of ESLint. The `.eslintrc` way of doing things means a variety of different merging strategies (cascading, extending, overriding), loading strategies (for plugins, processors, parsers), and naming conventions (`eslint-config-*`, `eslint-plugin-*`).
+
+Although `.eslintrc` has proven to be a very robust format, it is also limited and unable to easily accommodate feature requests that the community favors, such as:
+
+1. [bundling plugins with configs](https://github.com/eslint/eslint/issues/3458)
+1. [specifying ignore patterns in the config](https://github.com/eslint/eslint/issues/10891)
+1. [specifying `--ext` information in the config](https://github.com/eslint/eslint/issues/11223)
+1. [using `extends` in `overrides`](https://github.com/eslint/eslint/issues/8813) 
+
+The only reason that these requests are difficult to implement in ESLint is because of how complex the `.eslintrc` configuration format is. Any changes made to any part of `.eslintrc` processing end up affecting millions of ESLint installations, so we have ended up stuck. The complicated parts of `.eslintrc` include:
+
+1. Resolution behavior of modules (`plugins`, `parser`, `extends`)
+1. Merging of cascading config files in a directory structure
+1. Merging of config files via `extends`
+1. Overriding of configuration via `overrides`
+
+All of these complexities arise from `.eslintrc` format of describing what should happen rather than how it should happen. We keep trying to anticipate how users want things to happen and guessing wrong has led to increased complexity.
+
+This RFC proposes a fresh start for configuration in ESLint that takes into account all of the requests we've received over the years. By simplifying and stripping configuration we can actually make a configuration system that is flexible enough to accomodate changes in the future and easy enough to implement that we won't be afraid to do so.
+
+## Detailed Design
+
+Design Summary:
+
+1. Introduce a new `.eslint.js` configuration file format
```

Comment:
One thing to consider - a lot of popular JS tooling projects have started using the `project-name.config.js`  filename format (including [Webpack](https://webpack.js.org/configuration/), [Babel](https://babeljs.io/docs/en/configuration#babelconfigjs), and [Prettier](https://prettier.io/docs/en/configuration.html)). Glancing at `.eslint.js` quickly, I think it's very easy to misread it as `.eslintrc.js` if that's what you're used to. `eslint.config.js` would follow what is starting to become a convention and could make it clearer that it's a new file format.

Reply:
Oh, I wasn't aware of that. I'm happy to change the proposal to `.eslint.config.js`. Great feedback.

Reply:
> I'm happy to change the proposal to `.eslint.config.js`.

Just to clarify, the suggestion from https://github.com/eslint/rfcs/pull/9#discussion_r249959649 and https://github.com/eslint/rfcs/pull/9#discussion_r249968780 was to call the config `eslint.config.js` rather than `.eslint.config.js` (i.e. no leading dot). I'm happy to discuss other options but I'm not sure whether the leading dot here was a typo or an alternative suggestion.

Reply:
Matches what cosmiconfig looks for as well https://github.com/davidtheclark/cosmiconfig/blob/master/README.md

Reply:
Oops, I misread that. I'll change to `eslint.config.js`. Thanks for catching that.